src/main/kotlin/org/stellar/walletsdk/Anchor.kt

```diff
@@ -0,0 +1,172 @@
+   *
+   * @param assetIssuer Stellar address of the asset issuer (anchor)
+   *
+   * @return TOML file content
+   */
+  suspend fun getInfo(assetIssuer: String): Map<String, Any> {
+    return CoroutineScope(Dispatchers.IO)
+      .async {
+        val toml = StellarToml(stellarAddress = assetIssuer, server, httpClient)
+
+        return@async toml.getToml()
+      }
+      .await()
+  }
+
+  /**
+   * Available anchor services and information about them. For example, limits, currency, fees,
+   * payment methods.
+   *
+   * @param serviceUrl URL where `/info` endpoint is hosted
+   *
+   * @return a list of available anchor services
+   *
+   * @throws [NetworkRequestFailedException] if network request fails
+   * @throws [InvalidAnchorServiceUrl] if provided service URL is not a valid URL
+   */
+  suspend fun getServicesInfo(serviceUrl: String): AnchorServiceInfo {
+    try {
+      val url = serviceUrl.toHttpUrl()
+      val urlBuilder = HttpUrl.Builder().scheme(url.scheme).host(url.host).port(url.port)
+
+      url.pathSegments.forEach { urlBuilder.addPathSegment(it) }
+      urlBuilder.addPathSegment("info")
+
+      val infoUrl = urlBuilder.build().toString()
+
+      val request = Request.Builder().url(infoUrl).build()
+      val gson = GsonUtils.instance!!
+
+      return CoroutineScope(Dispatchers.IO)
+        .async {
+          httpClient.newCall(request).execute().use { response ->
+            if (!response.isSuccessful) throw NetworkRequestFailedException(response)
+
+            val infoResponse = response.body!!.charStream()
+            return@async gson.fromJson(infoResponse, AnchorServiceInfo::class.java)
+          }
+        }
+        .await()
+    } catch (e: Exception) {
+      throw InvalidAnchorServiceUrl(e)
+    }
+  }
+
+  /**
+   * Interactive deposit flow using
+   * [SEP-24](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0024.md).
+   *
+   * @param accountAddress Stellar address of the account
+   * @param assetCode Asset code to use
+   * @param assetIssuer Asset issuer to use
+   * @param memoId optional memo ID to distinguish the account
+   * @param clientDomain optional domain hosting stellar.toml file containing `SIGNING_KEY`
+   * @param extraFields Additional information to pass to the anchor
+   * @param walletSigner interface to define wallet client and domain (if using `clientDomain`)
+   * signing methods
+   *
+   * @return response object from the anchor
+   *
+   * @throws [AssetNotAcceptedForDepositException] if asset is not accepted for deposits
+   * @throws [AssetNotEnabledForDepositException] if asset is not enabled for deposits by the anchor
+   * @throws [NetworkRequestFailedException] if network request fails
+   */
+  suspend fun getInteractiveDeposit(
+    accountAddress: String,
+    assetCode: String,
+    assetIssuer: String,
+    memoId: String? = null,
+    clientDomain: String? = null,
+    extraFields: Map<String, Any>? = null,
```

Comment:
Jus curious because I was having trouble finding mention of it in the SEP: do we have any idea of what `extraFields` might be?